src/lib/tls/tls_session.cpp

```diff
@@ -138,39 +175,26 @@ Session_Summary::Session_Summary(const Server_Hello_13& server_hello,
 
    // In TLS 1.3 the key exchange algorithm is not negotiated in the ciphersuite
    // anymore. This provides a compatible identifier for applications to use.
-   m_kex_algo = kex_method_to_string([&] {
+
+   const auto group = [&]() -> std::optional<Named_Group> {
       if(psk_used() || was_resumption()) {
          if(const auto keyshare = server_hello.extensions().get<Key_Share>()) {
-            const auto group = keyshare->selected_group();
-            if(group.is_dh_named_group()) {
-               return Kex_Algo::DHE_PSK;
-            } else if(group.is_ecdh_named_curve() || group.is_x25519() || group.is_x448()) {
-               return Kex_Algo::ECDHE_PSK;
-            } else if(group.is_pure_kyber()) {
-               return Kex_Algo::KEM_PSK;
-            } else if(group.is_pqc_hybrid()) {
-               return Kex_Algo::HYBRID_PSK;
-            }
+            return keyshare->selected_group();
          } else {
-            return Kex_Algo::PSK;
+            return {};
          }
       } else {
          const auto keyshare = server_hello.extensions().get<Key_Share>();
          BOTAN_ASSERT_NONNULL(keyshare);
-         const auto group = keyshare->selected_group();
-         if(group.is_dh_named_group()) {
-            return Kex_Algo::DH;
-         } else if(group.is_ecdh_named_curve() || group.is_x25519() || group.is_x448()) {
-            return Kex_Algo::ECDH;
-         } else if(group.is_pure_kyber()) {
-            return Kex_Algo::KEM;
-         } else if(group.is_pqc_hybrid()) {
-            return Kex_Algo::HYBRID;
-         }
+         return keyshare->selected_group();
       }
+   }();
```

Comment:
Now, this could probably also just be:

```C++
   const auto group = [&]() -> std::optional<Named_Group> {
      if(const auto keyshare = server_hello.extensions().get<Key_Share>()) {
         return keyshare->selected_group();
      } else {
         return {};
      }
   }();
```

This removes the assertion in the non-PSK case, but if the handshake made it all the way to this session summary, that won't stop any bug here anymore, I guess. Edit: also, there's an equivalent assertion in (`tls13_kex_to_string()`, currently line 124).

Reply:
I adjusted the lambda as you suggest but added an assert later in the function where we examine if `group` has a value or not, since if this situation ever did happen the eventual fatal error (wherever it would occur) would probably be confusing and difficult to debug without it.